contracts/PvpArenaNEW.sol

```diff
@@ -347,6 +349,78 @@ contract PvpArena is Initializable, AccessControlUpgradeable {
         _duelQueue.add(attackerID);
     }
 
+    /// @dev allows a player to withdraw their ranking earnings
+    function withdrawRankedRewards() external {
+        uint256 amountToTransfer = _rankingRewardsByPlayer[msg.sender];
+
+        if (amountToTransfer > 0) {
+            _rankingRewardsByPlayer[msg.sender] = 0;
+
+            skillToken.safeTransfer(msg.sender, amountToTransfer);
+        }
+    }
+
+    /// @dev restarts ranked season
+    function restartRankedSeason() public restricted {
+        uint256[] memory duelQueue = getDuelQueue();
+
+        if (duelQueue.length > 0) {
+            performDuels(duelQueue);
+        }
+
+        // Note: Loops over 15 tiers. Should not be reachable anytime in the foreseeable future.
+        for (uint8 i = 0; i <= 15; i++) {
+            if (_topRankingCharactersByTier[i].length == 0) {
+                continue;
+            }
+
+            uint256 difference;
+
+            if (_topRankingCharactersByTier[i].length <= prizePercentages.length) {
+                difference = prizePercentages.length - _topRankingCharactersByTier[i].length;
+            } else {
+                difference = 0;
+            }
+
+            // Note: If there are less players than top positions, excess is transferred to top 1.
+            if (_topRankingCharactersByTier[i].length < prizePercentages.length) {
+                uint256 excessPercentage;
+                address topOnePlayer = characters.ownerOf(_topRankingCharactersByTier[i][0]);
+
+                // Note: We accumulate excess percentage.
+                for (
+                    uint256 j = prizePercentages.length - difference;
+                    j < prizePercentages.length;
+                    j++
+                ) {
+                    excessPercentage = excessPercentage.add(
+                        prizePercentages[j]
+                    );
+                }
+
+                // Note: We assign excessive rewards to top 1 player.
+                _rankingRewardsByPlayer[
+                    topOnePlayer
+                ] = _rankingRewardsByPlayer[topOnePlayer].add(
+                    (_rankingsPoolByTier[i].mul(excessPercentage)).div(100)
+                );
+            }
+
+            // Note: We assign rewards normally to all possible players.
```

Comment:
No need to prefix all comments with `Note:` all of a sudden 😂

Reply:
👁️